test/CreateClassFileTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace AntidotTest\DevTools\Application\Service;
+
+use Antidot\DevTools\Application\Service\CreateClassFile;
+use PHPUnit\Framework\TestCase;
+use RuntimeException;
+
+class CreateClassFileTest extends TestCase
+{
+    private CreateClassFile $createClassFile;
+
+    private string $classDir;
+
+    private string $className = 'Test';
+
+    private string $content = "<?php class Test{}\n";
+
+    protected function setUp(): void
+    {
+        $this->createClassFile = new CreateClassFile();
+        $this->classDir = __DIR__ . '/fixtures';
+    }
+
+    public function testItShouldReturnClassFileDirectory(): void
+    {
+        $expected = $this->classDir . '/' . $this->className . '.php';
+
+        $this->assertSame($expected, $this->createClassFile->__invoke($this->classDir, $this->className, $this->content));
+    }
+
+    public function testShouldThrowRuntimeExceptionOnDirectoryWasNotCreated()
+    {
+        $this->expectException(RuntimeException::class);
+
+        $invalidDir = 'php://temp';
+        $this->createClassFile->__invoke($invalidDir, $this->className, $this->content);
+    }
+
+    public function testShouldThrowRuntimeExceptionOnFileIsAlreadyExisted()
+    {
+        $this->expectException(RuntimeException::class);
+
+        $existedClassName = 'ExistedClass';
+        $this->createClassFile->__invoke($this->classDir, $existedClassName, $this->content);
+    }
+
+    protected function tearDown(): void
```

Comment:
When this class unit testing has been finished, it will trigger this `tearDown` method.

And I tried to clean specific class file generated by `testItShouldReturnClassFileDirectory` method.

But it seems that `guardrails` reports issue about insecure file management..

How could I fix above issue?

Reply:
I suggest using https://github.com/bovigo/vfsStream library, to make filesystem tests. When I create this package I didn't know it. You can see examples in their wiki, and I have some tests implemented in the installer https://github.com/antidot-framework/installer/blob/issue-2/test/Template/Micro/FileStructureTest.php.

The test setup will look more something like, and you will not need the teardown:

```php
<?PHP
...

    protected function setUp(): void
    {
        vfsStream::setup('fixtures');
    }
```

Reply:
Sure. I've used `bovigo/vfsStream` and let me have the change :).

Reply:
@kpicaza, thanks for your replay and suggestion.

I almost forgot we can use `vfsStream` to accomplish file writing test.

I also use this to comeplete the `testItShouldReturnClassFileDirectory` test :).

Now the guardrails has been passed.